cw_bitcoin/lib/litecoin_wallet.dart

```diff
@@ -915,10 +950,31 @@ abstract class LitecoinWalletBase extends ElectrumWallet with Store {
         }
       }
 
-      if (tx2.mwebBytes != null && tx2.mwebBytes!.isNotEmpty) {
-        hasMwebInput = true;
-      }
+      // check if mweb inputs are used:
+      final coinTypeToSpendFrom = transactionCredentials.coinTypeToSpendFrom;
+
+      await updateUnspent();
+      final availableInputs = unspentCoins.where((utx) {
+        if (!utx.isSending || utx.isFrozen) {
+          return false;
+        }
+
+        switch (coinTypeToSpendFrom) {
+          case UnspentCoinType.mweb:
+            return utx.bitcoinAddressRecord.type == SegwitAddresType.mweb;
+          case UnspentCoinType.nonMweb:
+            return utx.bitcoinAddressRecord.type != SegwitAddresType.mweb;
+          case UnspentCoinType.any:
+            return true;
+        }
+      }).toList();
 
+      for (var input in availableInputs) {
+        if (input.bitcoinAddressRecord.type == SegwitAddresType.mweb) {
+          hasMwebInput = true;
+        }
+      }
```

Comment:
this will be always true in any case except peg-in, as long as the user has mweb funds, regardless of whether they will be used in this transaction or not

Reply:
we filter the availableInputs list to only include coins the user has selected:
```
if (!utx.isSending || utx.isFrozen) {
```

so this case should be covered, but this highlights that the variable should probably renamed to something like `selectedInputs` to be less confusing

Reply:
I guess you missed my point,
at the end, you are setting `hasMwebInput` to true, if you find an mweb input in the user's selected inputs, but that doesn't mean that the tx will contain it, like it's selected yeah, but the transaction might make it with only normal ltc inputs

ex.
let's say I have 3 unspent coins
1. segwit 0.5
2. mweb 0.3
3. mweb 0.2

and I have all of them selected
then I go ahead and send 0.4
with this logic, it will mark it as `hasMwebInput`, while the actual inputs that will be used in tx might be only the first one (segwit 0.5), so that means the tx doesn't have mweb inputs, but the bool will be true regardless

Reply:
oh I see
I'm not sure how to tell if we're using mweb inputs then
the CreateRequest response from mwebd doesn't list mweb inputs (only non-mweb ones) in tx2.inputs
and the transactionCredentials object only has the outputs